bundles/org.eclipse.mita.base/src/org/eclipse/mita/base/typesystem/types/AbstractType.xtend

```diff
@@ -87,8 +89,28 @@ abstract class AbstractType {
 		map[it.replaceProxies(system, resolve)];
 	}
 	
-	def AbstractType modifyNames((String) => String converter) {
+	def AbstractType modifyNames(NameModifier converter) {
 		map[it.modifyNames(converter)];
 	}
 	
+	public static abstract class Either<LEFT, RIGHT> {
```

Comment:
`Either` looks like a key/value pair to me, however I did not grasp its purpose by looking at its uses. Could you elaborate a bit why it's there (as compared to, say a `Pair<T, R>`)?

Reply:
I will. In brief, `Either` is the opposite of `Pair`, where pairs always have both values, either only has one of them.